### PR TITLE
Special preflight failure handling

### PR DESCRIPTION
**Pull request by DonLakeFlyer**
Replacement for #6436

### CODE REVIEW

**Comment by DonLakeFlyer** [comment]
As far as severity I think these should be errors. But the problem with that is that it causes QGC to pop error popups for them which is annoying as well. Especially with the special handling. Maybe the error popping code should filter these out as well to not pop and error box. Opinions?

File: src/QGCApplication.cc [hunk 1]
```diff
@@ -647,8 +647,8 @@ QObject* QGCApplication::_rootQmlObject()
 
 void QGCApplication::showMessage(const QString& message)
 {
-    // Special case hack for ArduPilot prearm messages. These show up in the center of the map, so no need for popup.
-    if (message.contains("PreArm:")) {
+    // PreArm messages are handled by Vehicle and shown in Map
+    if (message.contains("PreArm")) {
```
Comment (dagar) [inline src/QGCApplication.cc]: OR PREFLIGHT FAIL?
At the moment we also have a couple instances of "PREFLIGHT ADVICE".

Comment (DonLakeFlyer) [reply src/QGCApplication.cc]: Oh, wait I forgot that was there. I'll change that as well. Which in turn means these should be shown with high severity and QGC wille at them.

Comment (DonLakeFlyer) [reply src/QGCApplication.cc]: And I'll change to PREFLIGHT prefix only.

**Comment by dagar** [comment]
I'm going to test the QGC changes and play with different MAV_SEVERITY levels based on attempted arming or running the check periodically.

**Comment by dagar** [comment]
Questions
 - I'm seeing what text directly on the map. Is it supposed to be in a yellow box?
 - Should we also remove the prefix string (PreArm: or PREFLIGHT FAIL:) before display?
 - On the firmware side is it worth limiting these to a single failure message at a time? Displaying multiple doesn't seem to work very well.

**Comment by DonLakeFlyer** [comment]
> I'm seeing what text directly on the map. Is it supposed to be in a yellow box?

The exact text of the Status Text messaage shows up in a big font in the center of the map.

**Pull request closed without merging**